include/simgrid/s4u/actor.hpp

```diff
@@ -54,14 +54,14 @@ XBT_PUBLIC_CLASS Actor {
   
   /** Method to create a new actor, the template parameter must extend the class Actor. In addition  the inheritance must be public. */
   template<class ActorSubClass> static ActorSubClass *createActor(const char*name, s4u::Host *host, int argc, char **argv, double killTime) {
-     Actor *res = dynamic_cast<Actor*>(new ActorSubClass());
+     ActorSubClass *res = new ActorSubClass();
      xbt_assert(res,"The actor class must extends the class simgrid::s4u::Actor.");
      
-     res->pimpl_ = simcall_process_create(name, s4u_actor_runner, static_cast<void*>(res), host->name().c_str(), killTime, argc, argv, NULL/*properties*/,1);
+     static_cast<Actor*>(res)->pimpl_ = simcall_process_create(name, s4u_actor_runner, static_cast<void*>(res), host->name().c_str(), killTime, argc, argv, NULL/*properties*/,1);
```

Comment:
In fact, I thunk there's still a (racy?) problem here because when the actor starts executing, the `pimpl_` field might still be `nullptr`.